libs/coin-modules/coin-evm/src/logic.ts

```diff
@@ -407,3 +408,23 @@ export const safeEncodeEIP55 = (addr: string): string => {
     return addr;
   }
 };
+
+/**
+ * Similar to mergeAccount but used to keep previous data we can't fetch on chain
+ */
+// logic.ts
+export const createSwapHistoryMap = (
+  initialAccount: Account | undefined,
+): Map<TokenCurrency, TokenAccount["swapHistory"]> => {
+  if (!initialAccount?.subAccounts) return new Map();
+
+  const swapHistoryMap = new Map<TokenCurrency, TokenAccount["swapHistory"]>();
+  for (const subAccount of initialAccount.subAccounts) {
+    if (subAccount.type === "TokenAccount") {
+      // <-- not even necessary anymore since we removed the `SubAccount` type now
+      swapHistoryMap.set(subAccount.token, subAccount.swapHistory);
+    }
+  }
```

Comment:
```suggestion
  for (const subAccount of initialAccount.subAccounts) {
    swapHistoryMap.set(subAccount.token, subAccount.swapHistory);
  }
```

Can now be removed since `Account.subAccounts` is now only `TokenAccounts[]` 
https://github.com/LedgerHQ/ledger-live/blob/develop/libs/ledgerjs/packages/types-live/src/account.ts#L141